cmd/auth-provider-gcp/app/getcredentials.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/cobra"
+	utilnet "k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/cloud-provider-gcp/cmd/auth-provider-gcp/credentialconfig"
+	"k8s.io/cloud-provider-gcp/cmd/auth-provider-gcp/plugin"
+	klog "k8s.io/klog/v2"
+	"net/http"
+)
+
+var (
+	authFlow string
+)
+
+const (
+	gcrAuthFlow             = "gcr"
+	dockerConfigAuthFlow    = "dockercfg"
+	dockerConfigURLAuthFlow = "dockercfg-url"
+)
+
+// NewGetCredentialsCommand returns a cobra command that retrieves auth credentials after validating flags.
+func NewGetCredentialsCommand() (*cobra.Command, error) {
+	cmd := &cobra.Command{
+		Use:   "get-credentials",
+		Short: "Get authentication credentials",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			klog.Infof("get-credentials %s", authFlow)
+			transport := utilnet.SetTransportDefaults(&http.Transport{})
+			var provider credentialconfig.DockerConfigProvider
+			if authFlow == gcrAuthFlow {
```

Comment:
Personally I prefer switches to multiple else blocks. Also we should not assume that if authFlow != gcrAuthFlow and authFlow != dockerConfigAuthFlow that it equals dockerConfigURLAuthFlow. We should expressly match that case and then have a final else (or default) block which gives an error for unrecognized authflow.

Reply:
Gotcha, done